### PR TITLE
Update JSDoc comments to add missing parameters

### PR DESCRIPTION
**Pull request by whymarrh**
This PR updates a few of the JSDoc comments that were incorrectly listing the arguments of the functions they were documenting.

### CODE REVIEW

**Comment by whymarrh** [comment]
As an aside: is there a way to add an extra ESLint rule to `standard`? From the docs:

> If you really want to configure hundreds of ESLint rules individually, you can always use `eslint` directly with [eslint-config-standard](
https://github.com/standard/eslint-config-standar)d to layer your changes on top. [`standard-eject`](
https://github.com/josephfrazier/standard-eject) can help you migrate from `standard` to `eslint` and `eslint-config-standard`.

Would the team be open to ejecting from standard to add, for example, [`valid-jsdoc`](https://eslint.org/docs/rules/valid-jsdoc)?

File: lib/bloom.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const byteSize = 256
  * drops leading zeroes from a buffer
  * @function dropLeadingZeroes
  * @param {Buffer} buff
+ * @returns {Buffer} a slice of the buffer starting at the first non-zero entry
```
Comment (holgerd77) [inline lib/bloom.js]: Ok.

File: lib/bloom.js [hunk 2]
```diff
@@ -32,7 +33,7 @@ var Bloom = module.exports = function (bitvector) {
 /**
  * adds an element to a bit vector of a 64 byte bloom filter
  * @method add
- * @param {Buffer} element
+ * @param e - the element
```
Comment (holgerd77) [inline lib/bloom.js]: I am not really getting the change here. This calls ``dropLeadingZeroes()`` which requires a Buffer.

File: lib/bloom.js [hunk 3]
```diff
@@ -48,9 +49,10 @@ Bloom.prototype.add = function (e) {
 }
 
 /**
- * checks if an element is in the blooom
+ * checks if an element is in the bloom
  * @method check
- * @param {Buffer} element
+ * @param e - the element
```
Comment (holgerd77) [inline lib/bloom.js]: Same here.

File: lib/bloom.js [hunk 4]
```diff
@@ -69,9 +71,10 @@ Bloom.prototype.check = function (e) {
 }
 
 /**
- * checks if multple topics are in a bloom
- * @method check
- * @param {Buffer} element
+ * checks if multiple topics are in a bloom
+ * @method multiCheck
+ * @param {Buffer} topics
+ * @returns {boolean} Returns {@code true}
```
Comment (holgerd77) [inline lib/bloom.js]: Not sure how much consistency we have with this in the code in general, but we should see that we stay with one convention using capital letter at the explanatory comment ("Returns...") or not.

File: lib/opFns.js [hunk 5]
```diff
@@ -799,6 +799,7 @@ function trap (err) {
 /**
  * Subtracts the amount needed for memory usage from `runState.gasLeft`
  * @method subMemUsage
+ * @param runState
```
Comment (holgerd77) [inline lib/opFns.js]: We normally have the convention to just use ``{Object}`` as a type if this is the most we can get 😄 (see other documentation in the library). If this is not wrong per se (I am not a super-comment-syntax expert) I would like to keep here with this as well.

File: lib/opFns.js [hunk 6]
```diff
@@ -829,6 +830,7 @@ function subMemUsage (runState, offset, length) {
  * a string is instead returned. The function also subtracts the amount of
  * gas need for memory expansion.
  * @method memLoad
+ * @param runState
```
Comment (holgerd77) [inline lib/opFns.js]: Same here.

File: lib/opFns.js [hunk 7]
```diff
@@ -858,6 +860,7 @@ function memLoad (runState, offset, length) {
  * Stores bytes to memory. If an error occurs a string is instead returned.
  * The function also subtracts the amount of gas need for memory expansion.
  * @method memStore
+ * @param runState
```
Comment (holgerd77) [inline lib/opFns.js]: Same here.

File: lib/opFns.js [hunk 8]
```diff
@@ -829,6 +830,7 @@ function subMemUsage (runState, offset, length) {
  * a string is instead returned. The function also subtracts the amount of
  * gas need for memory expansion.
  * @method memLoad
+ * @param runState
  * @param {BN} offset where to start reading from
  * @param {BN} length how far to read
  * @return {Buffer|String}
```
Comment (holgerd77) [inline lib/opFns.js]: Is it ``return`` or ``returns``?

File: lib/runBlockchain.js [hunk 9]
```diff
@@ -4,6 +4,7 @@ const async = require('async')
  * processes blocks and adds them to the blockchain
  * @method onBlock
  * @param blockchain
+ * @param {Function} cb the callback
```
Comment (holgerd77) [inline lib/runBlockchain.js]: Ok.

File: lib/runCall.js [hunk 10]
```diff
@@ -10,6 +10,7 @@ const ERROR = exceptions.ERROR
  * runs a CALL operation
  * @method runCall
  * @param opts
+ * @param cb the callback
```
Comment (holgerd77) [inline lib/runCall.js]: This should be at the end of the ``opts.xxx`` param values (or again, if this is generally wrong, changed everywhere for consistency).

File: lib/stateManager.js [hunk 11]
```diff
@@ -237,8 +237,8 @@ proto.getStateRoot = function (cb) {
 }
 
 /**
- * @param {Set} address
- * @param {cb} function
+ * @param {Set} addresses a set of addresses
+ * @param {Function} cb the callback
```
Comment (holgerd77) [inline lib/stateManager.js]: Ok.

File: lib/bloom.js (hunk 2 above)
Comment (whymarrh) [reply lib/bloom.js]: Yup, should've kept the type, I'll add that back. The main change is the param name which was incorrect (i.e. `s/element/e/`).

File: lib/opFns.js (hunk 5 above)
Comment (whymarrh) [reply lib/opFns.js]: Yup, I'll add the type in

File: lib/opFns.js (hunk 8 above)
Comment (whymarrh) [reply lib/opFns.js]: Either is valid, I believe `@returns` is more prevalent in the codebase from what I see currently

File: lib/runCall.js (hunk 10 above)
Comment (whymarrh) [reply lib/runCall.js]: Sure thing

File: lib/bloom.js (hunk 4 above)
Comment (whymarrh) [reply lib/bloom.js]: I had noticed that a majority of the `@returns` clauses started with "Returns" so I'm following that—I can update the comments for consistency in a separate PR if we want?

**Comment by whymarrh** [comment]
> I think moving from `standard` to `eslint` directly is not an option regarding the added maintenance work. If there is a simple way just to add one rule or the other to `standard` I would be eventually open to it, but I guess that just contradicts the original idea of the library?

Yeah from what I can tell the standardjs setup is quite specific and, as a result, somewhat inflexible. Not moving to ESLint is fine, just thought I'd bring it up as an option for more static analysis.

File: lib/bloom.js (hunk 4 above)
Comment (holgerd77) [reply lib/bloom.js]: Ah, I think that's overblown for now. If you do an update with the other changes that should suffice for now.

**Comment by whymarrh** [comment]
Thanks for the review, I'll update this today

**Comment by holgerd77** [comment]
Can this be updated so we can merge? That would be great, PR like this with both a broad scope and low-level changes tend to otherwise cause a disproportionate amount of additional work keeping things up-to-date...

**Comment by whymarrh** [comment]
> There are some ``@return`` -> ``@returns`` cases which needs to be updated. The following normally wouldn't be a blocker, but if you update please also remove the ``-``  from the ``@param {Buffer} e - the element to check`` since this introduces a minor inconsistency in the format.

Updated

**Pull request merged**